src/plugins/hexnumber/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+include (LibAddMacros)
```

Comment:
Please format this file with the script [`reformat-cmake`](https://github.com/ElektraInitiative/libelektra/blob/master/doc/CODING.md#cmake-guidelines).

Reply:
Thanks for noting, we should enforce that with the buildserver. Isn't this just a matter of installing two tools?

Ideally the build server should directly tell that something like that is wrong.

@ingwinlu What do you think?

Reply:
> Thanks for noting, we should enforce that with the buildserver.

We already do that in the [Travis Linux builds](https://travis-ci.org/ElektraInitiative/libelektra/jobs/380362048#L3163).

> Isn't this just a matter of installing two tools?

[Yes](https://github.com/ElektraInitiative/libelektra/blob/master/.travis.yml#L139-L140).

Reply:
knock yourself out :D

Reply:
@sanssecours It seems you need to manually install libyaml as well as [PyYAML](https://github.com/yaml/pyyaml) for the `reformat-cmake` script to work. This should probably be reflected in our documentation. Especially because otherwise the script throws an error for each CMake file and also deletes the contents of all these files.

Reply:
> @sanssecours It seems you need to manually install libyaml as well as [PyYAML](https://github.com/yaml/pyyaml) for the `reformat-cmake` script to work. 

As far as I know `cmake-format` has no dependencies:

```sh
pip show cmake-format
#> Name: cmake-format
#> Version: 0.3.6
#> Summary: Can format your listfiles so they don't look like crap
#> Home-page: https://github.com/cheshirekow/cmake_format
#> Author: Josh Bialkowski
#> Author-email: josh.bialkowski@gmail.com
#> License: UNKNOWN
#> Location: /Users/rene/.pyenv/versions/3.6.5/lib/python3.6/site-packages
#> Requires:
#> Required-by:
```

. 

> This should probably be reflected in our documentation. Especially because otherwise the script throws an error for each CMake file and also deletes the contents of all these files.

While [`reformat-cmake`](https://github.com/ElektraInitiative/libelektra/blob/master/scripts/reformat-cmake) is certainly not perfect, it checks for the binaries `cmake-format` and `sponge` beforehand. If you did not install one of these tools it should just fail printing a (broken) error message. I tried this myself by uninstalling `cmake-format` and after that uninstalling `moreutils`. Are you sure that the script deleted you CMake files. If so, can you please describe steps to reproduce the problem?

Reply:
> Are you sure that the script deleted you CMake files?

It did not delete the files themselves, just their contents. I think this is because stdout of `cmake-format` is piped back into the CMake file, but stdout is empty because of the error.

> As far as I know cmake-format has no dependencies

I also looked at the `cmake-format` github page and could not find any mention of the dependency. But [here](https://github.com/cheshirekow/cmake_format/blob/2e2aff2abaa5040bb89e6697d15ecbd628d37b07/cmake_format/__main__.py#L101) the module `yaml` is imported, which is the cause of the error. 

The problem could probably be solved by using the `json` or `python` format for the `.cmake-format` file, because these two modules are part of the python standard library.


If you still want me to, I can try to find out how to reproduce the error.

Reply:
Thank you for the detailed description. I opened pull request #2011, which should fix the problem by checking the exit code of `cmake-format CMakeLists.txt` before it updates any file.

> The problem could probably be solved by using the `json` or `python` format for the `.cmake-format` file, because these two modules are part of the python standard library.

That might even be a better solution, than the one I came up with. Anyway, since I prefer the human readability of YAML over JSON or the Python config style, I did not change the format of the config file.